server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -324,12 +318,10 @@ private synchronized DocumentMapper internalMerge(DocumentMapper mapper, MergeRe
             fullPathObjectMappers = Collections.unmodifiableMap(fullPathObjectMappers);
         }
 
-        if (newMapper != null) {
-            this.mapper = newMapper;
-        }
+        this.mapper = newMapper;
```

Comment:
newMapper is always not null, we call methods on it: https://github.com/crate/crate/pull/14195/files#diff-32c1e36544e0c0ef775f9e26576ef7df0504cc05466e513ba7b143947c234784R288